documentation/movies.workspace.adoc

```diff
@@ -0,0 +1,731 @@
+
+WITH TomH as a
+MATCH (a)-[:ACTED_IN]->(m)<-[:DIRECTED]-(d) RETURN a,m,d LIMIT 10;
+----
+
+In the next step, you will look for nodes and their properties.
+
+== Find nodes
+
+To start, can you find the actor Tom Hanks?
+
+Run the following query:
+
+[source,cypher]
+----
+MATCH (tom {name: "Tom Hanks"})
+RETURN tom
+----
+
+Next, can you find a movie called _Cloud Atlas_?
+
+[%collapsible]
+.Reveal the solution
+====
+[source,cypher]
+----
+MATCH (movie {title:"Cloud Atlas"})
+RETURN movie
+----
+====
+
+Similarly, you can find all the people (nodes with the `Person` label) in the graph, but limit the returned number to ten:
+
+[source,cypher]
+----
+MATCH (people:Person)
+RETURN people.name LIMIT 10
+----
+
+[TIP]
+====
+Try to remove the property key `.name` from the RETURN clause and see the difference.
+(`RETURN people LIMIT 10`)
+====
+
+If you want to find movies released in a specific timespan, you can add conditions by using `WHERE`:
+
+[source,cypher]
+----
+MATCH (nineties:Movie) WHERE nineties.released >= 1990 AND nineties.released < 2000
+RETURN nineties.title
+----
+
+In the next step you will look for patterns in the graph.
+
+== Find patterns
+
+A pattern in the graph is a specific arrangement of nodes and relationships that can be matched in the graph.
+To find which movies Tom Hanks have acted in, specify the pattern as follows:
+
+[source,cypher]
+----
+MATCH (tom:Person {name: "Tom Hanks"})-[:ACTED_IN]->(tomHanksMovies)
+RETURN tom,tomHanksMovies
+----
+
+Can you specify a pattern to find who directed _Cloud Atlas_?
+Hint: The relationship between a person who is a director and the movie they directed is `:DIRECTED`.
+
+[%collapsible]
+.Reveal the solution
+====
+[source,cypher]
+----
+MATCH (cloudAtlas {title: "Cloud Atlas"})<-[:DIRECTED]-(directors)
+RETURN directors.name
+----
+====
+
+If you extend the length of the pattern, you can find co-actors:
+
+[source,cypher]
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors)
+RETURN coActors.name
+----
+
+You can return _all_ properties if you are not sure exactly what you are looking for:
+
+[source,cypher]
+----
+MATCH (people:Person)-[relatedTo]-(:Movie {title: "Cloud Atlas"})
+RETURN people.name, Type(relatedTo), relatedTo
+----
+
+This query returns a list of Person-nodes that are connected to the Movie-node _Cloud Atlas_, the relationship type, and _all_ the properties on the relationship.
+
+In the next step, you are going to prove whether the Hollywood version of six degrees of separation is true in your dataset.
+
+== Six degrees of Kevin Bacon
+
+You may have heard about the concept that any two people on Earth are six or fewer connections apart.
+In Hollywood, this is said to be true for Kevin Bacon and any other actor.
+
+The following query lets you see all movies and actors up to six hops away from Kevin Bacon:
+
+[source,cypher]
+----
+MATCH (bacon:Person {name:"Kevin Bacon"})-[*1..6]-(hollywood)
```

Comment:
`bacon` -> leave off, we don't use it in the query
`hollywood` -> `person`

Reply:
✅